pkg/controller/identity/iampolicy/controller.go

```diff
@@ -99,7 +99,14 @@ func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.E
 	}
 
 	if meta.GetExternalName(cr) == "" {
-		return managed.ExternalObservation{}, nil
+		// If external name not set there is still a change it may already exist
+		// Try to get the policy by name
+		policyArn, _ := e.getPolicyArnByName(ctx, cr.Spec.ForProvider.Name)
```

Comment:
If there are thousands of policies, this might be a bit expensive? It might be cheaper to do `arn:aws:iam::+AccountID+:policy+Path+.Name`, but maybe it's difficult because we don't have easy access to the account number?

Reply:
@chlunde - all good points.  I am going to see if [`GetCallerIdentity`](https://docs.aws.amazon.com/STS/latest/APIReference/API_GetCallerIdentity.html) can be used instead.  If that call is able to return enough values to construct the ARN then the ARN should be able to used to make one single api call to see if the policy exists.